### PR TITLE
[test] Add network limiting e2e test

### PR DESCRIPTION
**Pull request by WVerlaek**
## Description

Add test for network limiting.

Works by:
* installing an agent in ws-daemon, with a new `GetNftRulesets` RPC
* this agent then finds the workspace container's `pid`
* then the `ring0` child process of the container process
* enters the network namespace of the ring0 process
* checks `nft list ruleset` and verifies rate limiting is present

Took a while to get this working 😄 

## Related Issue(s)

Fixes WKS-223

## How to test

```
cd test
go test -v ./... -run TestNetworkLimiting
```

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: test/pkg/integration/integration.go [hunk 1]
```diff
@@ -208,6 +209,11 @@ func (r *RpcClient) Call(serviceMethod string, args any, reply any) error {
 	for i := 0; i < connectFailureMaxTries; i++ {
 		if r != nil {
 			if err = r.client.Call(serviceMethod, args, reply); err != nil {
+				log.Warnf("rpc call %s failed (attempt %d): %v", serviceMethod, i, err)
+				if i == connectFailureMaxTries-1 {
+					return err
+				}
```
Comment (WVerlaek) [inline test/pkg/integration/integration.go:215]: on the final attempt, return the RPC error, otherwise the error gets swallowed below

**Comment by kylos101** [comment]
@Furisto could you help @WVerlaek with a review on this one? :pray:

**Comment by kylos101** [comment]
@WVerlaek the build is failing as a heads up

**Review by Furisto — APPROVED** [review]

**Pull request merged**